src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -610,6 +612,23 @@ def _job_clusters(self, names: set[str]):
             )
         return clusters
 
+    def _job_log_task(self, tasks: list[Task], remote_wheel: str) -> jobs.Task:
+        task = Task(
+            task_id=len(tasks),
+            workflow=tasks[0].workflow,
+            name="parse-logs",
```

Comment:
@nfx : The log task is appended above for all workflows. However, [this causes a `KeyError` at runtime as the `"parse-logs"` tasks does not exists in `_TASKS`](https://github.com/databrickslabs/ucx/blob/dd6b84e18cca71970553844c2deaa45e30c3e583/src/databricks/labs/ucx/framework/tasks.py#L234). 

In [my other comment](https://github.com/databrickslabs/ucx/pull/1272#issuecomment-2037043724), I try the problem of this dynamically inserted task not fitting the regular `task` definition in the `tasks.py` module. 

A solution could be to add an extra endpoint for the `logs` task only. This would also allow us to pass the `parent_run_id` to the log task, which is required to identify the logs: `self._log_path = install_dir / "logs" / self._workflow / f"run-{self._workflow_run_id}"`

Reply:
> A solution could be to add an extra endpoint for the logs task only.

yep. add this to runtime.py, it should unblock you. If some part of the system complains that parse_logs belongs to more than one workflow - add an `if` statement.

```
@task('...')
def parse_logs(...):
   ...
```